docs/guides/web/nginx-mainline.md

```diff
@@ -0,0 +1,282 @@
+```
+
+Let’s break this down: 
+
+* The `-–permanent` flag tells the firewall to make sure this configuration is used every time the firewall is restarted, and when the server itself is restarted. 
+* `–-zone=public` tells the firewall to take incoming connections to this port from everyone.
+* Lastly, `–-add-port=80/tcp` tells the firewall to accept incoming connections over port 80, as long as they’re using the Transmission Control Protocol, which is what you want in this case.
+
+To repeat the process for SSL/HTTPS traffic, just run the command again, and change the number.
+
+```bash
+sudo firewall-cmd --permanent --zone=public --add-port=443/tcp
+```
+
+These configurations won’t take effect until you force the issue. To do that, tell *firewalld* to relead its configurations, like so:
+
+```bash
+sudo firewall-cmd --reload
+```
+
+Now, there’s a very small chance that this won’t work. In those rare cases, make *firewalld* do your bidding with the old turn-it-off-and-turn-it-on-again.
+
+```bash
+systemctl restart firewalld
+```
+
+To make sure the ports have been added properly, run `firewall-cmd --list-all`. A properly-configured firewall will look a bit like this (I have a few extra ports open on my local server, ignore them):
+
+```bash
+public (active)
+  target: default
+  icmp-block-inversion: no
+  interfaces: enp9s0
+  sources:
+  services: cockpit dhcpv6-client ssh
+  ports: 81/tcp 444/tcp 15151/tcp 80/tcp 443/tcp
+  protocols:
+  forward: no
+  masquerade: no
+  forward-ports:
+  source-ports:
+  icmp-blocks:
+  rich rules:
+```
+
+And that should be everything you need, firewall-wise.
+
+*Now* you should be able to see a web page that looks something like this:
+
+![The Nginx welcome page](nginx/images/welcome-nginx.png)
+
+It’s not much at all, but it means the server is working.
+
+## Creating a Server User and Changing the Website Root Folder
+
+While you *can* just drop your website into the default directory and go (and this might be fine for *Nginx* when it’s running inside a container, or on a test/development server), it’s not what we call best practice. Instead, it’s a good idea to create a specific Linux user on your system for your website, and put your website files in a directory made just for that user. 
+
+If you want to build multiple websites, it’s actually a good idea to create multiple users and root directories, both for the sake of organization and the sake of security. 
+
+In this guide, I’m going to have just the one user: a handsome devil named “www”. We’re going to put all of his website files under a directory in its home folder: `/home/www/`. You can actually put the folder anywhere you want, but on a dedicated server machine, using home folders makes perfect sense.
+
+### Creating the User
+
+First, we make the folder we’re going to use:
+
+```bash
+sudo mkdir /home/www/
+```
+
+Then, we create the user:
+
+```bash
+sudo adduser -g 'Nginx www user' -h /home/www/ www
```

Comment:
This user should be a system user and not having a bash (/bin/nologin for example).

Reply:
Hmmm. I was taking use cases like client hosting accounts into consideration. Also, is there a specific change to the command to make it a system user?

Maybe we can talk about both use cases.

Reply:
I think most of the time, ftp or sftp user to access the docroot are virtual (or should be).
We can keep the id of a user creation, even not a system one, but the discourse must be clear on security. It's a good way to do it, but it's not the most secure way.

Reply:
Okay, all good stuff to know.

Reply:
@EzequielBruni if you read over the vsftp guide that I wrote, this uses virtual users. Pretty cool!